src/command/FileCommands.cxx

```diff
@@ -224,12 +230,212 @@ handle_read_comments(Client &client, Request args, Response &r)
 	case LocatedUri::Type::ABSOLUTE:
 		return read_stream_comments(r, located_uri.canonical_uri);
 
-	case LocatedUri::Type::RELATIVE:
-		return read_db_comments(client, r, located_uri.canonical_uri);
 
-	case LocatedUri::Type::PATH:
-		return read_file_comments(r, located_uri.path);
-	}
+    case LocatedUri::Type::RELATIVE:
+        return read_db_comments(client, r, located_uri.canonical_uri);
+
+    case LocatedUri::Type::PATH:
+        return read_file_comments(r, located_uri.path);
+    }
+
+    gcc_unreachable();
+}
+
+InputStreamPtr
+find_stream_art(const char *directory)
+{
+    static const std::string art_names[] = {
+        "cover.png",
+        "cover.jpg",
+        "cover.tiff",
+        "cover.bmp"
+    };
+    
+    for(const std::string& name: art_names) {
+        const std::string art_file = PathTraitsUTF8::Build(directory, name.c_str());
+        
+        try {
+            Mutex mutex;
+            Cond cond;
+            
+            auto is = InputStream::OpenReady(art_file.c_str(), mutex, cond);
+            return is;
+        } catch (const std::exception &e) {
+            continue;
+        }
+    }
+    return nullptr;
+}
+
+static CommandResult
+read_stream_art(Response &r, const char *uri, uint64_t offset)
+{
+    std::string art_directory = PathTraitsUTF8::GetParent(uri);
+    InputStreamPtr is = find_stream_art(art_directory.c_str());
+    
+    if (is == nullptr) {
+        r.Error(ACK_ERROR_NO_EXIST, "No art file exists");
+        return CommandResult::ERROR;
+    }
+    
+    if (!is->IsSeekable()) {
```

Comment:
No, think `offset==0`, which requires no seeking ability at all. And it is the most common value. MPD should make a best effort to fulfill the client's request.
Denying the most common request just because the `InputStream` is not seekable, when seeking is not necessary, is somewhat dull.